### ISSUE TITLE
Support user defined cloning of volume data from existing disks.

### ISSUE THREAD

**Issue by govint** [vmware/docker-volume-vsphere#1855]
vDVS supports cloning a volume from an existing source volume.  This presently creates a complete copy of the source volume. To allow end users (dev typically) to clone just the data they need it should be possible to support a clone option that devs can use to provide a script of their own to clone specific data to a new target disk. The script would need to be able to take the source and destination paths as input and then be able to run and copy what data the dev needs in the cloned disk. The plugin would in this case create a new disk based on the options provided and then invoke the user script on the target disk as part of the _docker volume create_ command itself.

Use case is for scenarios where dev already has data in a volume and wants to share a subset of the data. Here, dev doesn't want to create a replica of the whole disk, instead just a copy of a subset of the data is to be cloned.

Another one is it allows devs to have the control to perform transformations on the data as its copied over to the target volume.

Basically, the new option will allow dev to integrate usage of the plugin and the docker volume commands into their data mgmt. workflows vs. making it a multi-step process where dev creates separate volumes and then mounts those and then copies over data, effectively separating mgmt of volumes and data handling by devs.

**Comment by pshahzeb** [comment]
Nice.
One customer had such a use case and resorted to [docker cp](https://docs.docker.com/engine/reference/commandline/cp/) command.
Would be good to see how multi-step process using docker cp is cut down if we introduce this feature.

**Comment by govint** [comment]
Docker cp helps copy data in or out of a container plus it won;t allow the devs to be able to make any modifications to the data the way they want to. There is a lot of leverage the dev can do with this proposed feature and it helps integrate storage ops (like creating a volume) into their data handling actions. Like, dev's app has collected data and they want to get a snap of that data into a new volume to be used by another app or for archiving, they can do this via this feature vs. multiple steps.

**Comment by shuklanirdesh82** [comment]
mustfix for atleast investigation task

**Comment by govint** [comment]
Estimate is only to investigate the usecases and verify how well or not users can leverage this capability.

**Comment by tusharnt** [comment]
@govint please update the results of your investigation.
